tests/scene/resources/test_image_texture.h

```diff
@@ -0,0 +1,225 @@
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_IMAGE_TEXTURE_H
+#define TEST_IMAGE_TEXTURE_H
+
+#include "scene/resources/image_texture.h"
+
+#include "tests/test_macros.h"
+#include "tests/test_utils.h"
+
+namespace TestImageTexture {
+const int default_height = 64;
+const int default_width = 64;
+const int default_channels = 3;
+const Image::Format default_format = Image::Format::FORMAT_RGB8;
+
+Ref<Image> create_test_image_base(int p_channels, Image::Format p_format) {
+	Vector<uint8_t> data;
+	data.resize(default_width * default_height * p_channels);
+
+	// This loop fills the data with image pixel values (RGBA format).
+	for (int y = 0; y < default_height; y++) {
+		for (int x = 0; x < default_width; x++) {
+			int offset = (y * default_width + x) * p_channels;
+			for (int c = 0; c < p_channels; c++) {
+				data.set(offset + c, 255);
+			}
+		}
+	}
+
+	return Image::create_from_data(default_width, default_height, false, p_format, data);
+}
+
+Ref<Image> create_test_image() {
+	return create_test_image_base(default_channels, default_format);
+}
+
+Ref<Image> create_test_image_with_alpha() {
+	return create_test_image_base(4, Image::FORMAT_RGBA8);
+}
+
+TEST_CASE("[ImageTexture] Constructor") {
+	// Create an ImageTexture object.
+	Ref<ImageTexture> imageTexture;
+	imageTexture.instantiate();
+
+	// Ensure that it's not null.
+	REQUIRE_FALSE(imageTexture.is_null());
+
+	// Verify default values are correct.
+	CHECK(imageTexture->get_width() == 0);
+	CHECK(imageTexture->get_height() == 0);
+	CHECK(imageTexture->get_format() == Image::FORMAT_L8);
+	CHECK_FALSE(imageTexture->has_alpha());
+}
+
+TEST_CASE("[ImageTexture][SceneTree] Set image") {
```

Comment:
```suggestion
TEST_CASE("[ImageTexture] Set image") {
```
Clean up these accidental leftover names

Reply:
`[SceneTree]` is actually not an accident - adding this name to the test name makes it so the `RenderingEngine` is created for the test, which happens here: https://github.com/godotengine/godot/blob/fe5b1c8d49313d63fbe91cb7cdf463e10fb86afa/tests/test_main.cpp#L223

There are a few other spots in the codebase where `[SceneTree]` is used to denote that the test uses the `RenderingEngine`. I'm not sure if that's standard or if those were accidents. Some examples:

* https://github.com/godotengine/godot/blob/fe5b1c8d49313d63fbe91cb7cdf463e10fb86afa/tests/core/input/test_input_event.h#L78
* https://github.com/godotengine/godot/blob/fe5b1c8d49313d63fbe91cb7cdf463e10fb86afa/tests/scene/test_code_edit.h#L40
* https://github.com/godotengine/godot/blob/fe5b1c8d49313d63fbe91cb7cdf463e10fb86afa/tests/scene/test_color_picker.h#L40
* https://github.com/godotengine/godot/blob/fe5b1c8d49313d63fbe91cb7cdf463e10fb86afa/tests/scene/test_node.h#L81

There are quite a few more cases. I do agree that this approach is a little messy though. I'm open to fixing it in this PR but I'm not sure what a better solution would be.

Reply:
My bad! Wasn't aware of this quirk, thank you for clarifying